CMakeLists.txt

```diff
@@ -106,36 +108,157 @@ set(AZURE_STORAGE_LITE_SOURCE
   src/blob/blob_client_wrapper.cpp
 )
 
-find_package(CURL REQUIRED)
-
-option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" OFF)
-
-find_package(Threads REQUIRED)
-pkg_search_module(UUID REQUIRED uuid)
-
-if (NOT ${USE_OPENSSL})
-  find_package(GnuTLS REQUIRED)
-  set(EXTRA_INCLUDE ${GNUTLS_INCLUDE_DIR})
-  set(EXTRA_LIBRARIES ${GNUTLS_LIBRARIES})
-else()
+if(APPLE)
+  option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" ON)
   add_definitions(-DUSE_OPENSSL)
+  find_package(CURL REQUIRED)
```

Comment:
Can we add a custom version of Curl as well? Could be a separate commit/PR, though.
We use a custom version curl since we inject extra security checks.